src/Rules/PasswordComplexity.php

```
@@ -0,0 +1,174 @@
+    public function __construct( ?Authenticatable $user = null )
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Determine if the validation rule passes.
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     */
+    public function passes( $attribute, $value ): bool
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$attribute'. (undefined)

(UnusedFormalParameter)